dds/DCPS/DiscoveryBase.h

```diff
@@ -2419,47 +2419,47 @@ namespace OpenDDS {
 #ifndef DDS_HAS_MINIMUM_BIT
     DCPS::ParticipantBuiltinTopicDataDataReaderImpl* part_bit()
     {
-      if (!bit_subscriber_.in())
+      if (!bit_subscriber_.value().in())
         return 0;
 
       DDS::DataReader_var d =
-        bit_subscriber_->lookup_datareader(BUILT_IN_PARTICIPANT_TOPIC);
+        bit_subscriber_.value()->lookup_datareader(BUILT_IN_PARTICIPANT_TOPIC);
       return dynamic_cast<ParticipantBuiltinTopicDataDataReaderImpl*>(d.in());
     }
 
     DCPS::ParticipantLocationBuiltinTopicDataDataReaderImpl* part_loc_bit()
     {
-      if (!bit_subscriber_.in())
+      if (!bit_subscriber_.value().in())
         return 0;
 
       DDS::DataReader_var d =
-        bit_subscriber_->lookup_datareader(DCPS::BUILT_IN_PARTICIPANT_LOCATION_TOPIC);
+        bit_subscriber_.value()->lookup_datareader(DCPS::BUILT_IN_PARTICIPANT_LOCATION_TOPIC);
       return dynamic_cast<ParticipantLocationBuiltinTopicDataDataReaderImpl*>(d.in());
     }
 
     DCPS::ConnectionRecordDataReaderImpl* connection_record_bit()
     {
-      if (!bit_subscriber_.in())
+      if (!bit_subscriber_.value().in())
         return 0;
 
       DDS::DataReader_var d =
-        bit_subscriber_->lookup_datareader(DCPS::BUILT_IN_CONNECTION_RECORD_TOPIC);
+        bit_subscriber_.value()->lookup_datareader(DCPS::BUILT_IN_CONNECTION_RECORD_TOPIC);
       return dynamic_cast<ConnectionRecordDataReaderImpl*>(d.in());
     }
 
     DCPS::InternalThreadBuiltinTopicDataDataReaderImpl* internal_thread_bit()
     {
-      if (!bit_subscriber_.in())
+      if (!bit_subscriber_.value().in())
         return 0;
 
       DDS::DataReader_var d =
-        bit_subscriber_->lookup_datareader(DCPS::BUILT_IN_INTERNAL_THREAD_TOPIC);
+        bit_subscriber_.value()->lookup_datareader(DCPS::BUILT_IN_INTERNAL_THREAD_TOPIC);
       return dynamic_cast<InternalThreadBuiltinTopicDataDataReaderImpl*>(d.in());
     }
 #endif /* DDS_HAS_MINIMUM_BIT */
 
       mutable ACE_Thread_Mutex lock_;
-      DDS::Subscriber_var bit_subscriber_;
+      ACE_Atomic_Op<ACE_Thread_Mutex, DDS::Subscriber_var> bit_subscriber_;
```

Comment:
One thing I should mention here is that the "right" fix for this is to fix TAO_Objref_Var_T to have some kind of synchronization around its pointer. But for now this seems to make ThreadSanitizer Happy.